### PR TITLE
Replace relrefs with docrefs in Agent tutorials

### PR DESCRIPTION
**Pull request by clayton-cornell**
#### PR Description

The relref shortcode is being deprecated in favour of the docs/reference shortcode. The new shortcode allows us to have more control over the references and links and makes sure that topics mounted into more than one project can have working links within that project.

This PR continues the process of migrating the Flow documentation to use the docs/reference shortcodes as preparation for mounting Flow topics into the Cloud docs

#### PR Checklist

- [ ] CHANGELOG.md updated
- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: docs/sources/flow/tutorials/filtering-metrics.md [hunk 1]
```diff
@@ -31,23 +31,35 @@ curl https://raw.githubusercontent.com/grafana/agent/main/docs/sources/flow/tuto
...
+Open the `relabel.river` file that was downloaded and change the name of the service to `api_server_v2`, then run `bash ./runt.sh relabel.river`. Allow Grafana Agent to run for two minutes, then navigate to the [Grafana][] page, and the new label will be updated. The old value `api_server` may still show up in the graph but hovering over the lines will show that that value stopped being scraped and was replaced with `api_server_v2`.
 
 ![Updated dashboard showing api_server_v2](/media/docs/agent/screenshot-grafana-agent-filtering-metrics-transition.png)
+
+
+[Docker]: https://www.docker.com/products/docker-desktop
+[Grafana]: http://localhost:3000/explore?orgId=1&left=%5B%22now-1h%22,%22now%22,%22Mimir%22,%7B%22refId%22:%22A%22,%22instant%22:true,%22range%22:true,%22exemplar%22:true,%22expr%22:%22agent_build_info%7B%7D%22%7D%5D
```
Comment (erikbaranowski) [inline docs/sources/flow/tutorials/filtering-metrics.md:57]: check this one. I'm assuming localhost isn't correct.

File: docs/sources/flow/tutorials/filtering-metrics.md [hunk 2]
```diff
@@ -31,23 +31,35 @@ curl https://raw.githubusercontent.com/grafana/agent/main/docs/sources/flow/tuto
...
+[relabel.river]: https://grafana.com/docs/agent/latest/flow/tutorials/assets/flow_configs/relabel.river
```
Comment (erikbaranowski) [inline docs/sources/flow/tutorials/filtering-metrics.md:58]: linking to latest instead of relative here

**Review by erikbaranowski — APPROVED** [review]
LGTM, a couple links to check in on

File: docs/sources/flow/tutorials/filtering-metrics.md (hunk 2 above)
Comment (clayton-cornell) [reply docs/sources/flow/tutorials/filtering-metrics.md:58]: @erikbaranowski aaaah another I missed. I get distracted and forget to triple check. I'll push a commit to fix this.

File: docs/sources/flow/tutorials/filtering-metrics.md (hunk 1 above)
Comment (clayton-cornell) [reply docs/sources/flow/tutorials/filtering-metrics.md:57]: @erikbaranowski I thought the same, but when you read in detail.. the tutorial has the reader set up a local instance, and in that case, localhost seems to make sense.

**Pull request merged**